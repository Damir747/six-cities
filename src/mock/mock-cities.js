const cities = {
  'Paris': { lat: 48.85661, lng: 2.351499, zoom: 13 },
  'Cologne': { lat: 50.938361, lng: 6.959974, zoom: 13 },
  'Brussels': { lat: 50.846557, lng: 4.351697, zoom: 13 },
  'Amsterdam': { lat: 52.37454, lng: 4.897976, zoom: 13 },
  'Hamburg': { lat: 53.550341, lng: 10.000654, zoom: 13 },
  'Dusseldorf': { lat: 51.225402, lng: 6.776314, zoom: 13 },
};

export default cities;
