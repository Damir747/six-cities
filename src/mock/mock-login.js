const loginName = 'Oliver.conner@gmail.com';
export default loginName;
