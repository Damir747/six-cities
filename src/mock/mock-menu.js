const menuUpArray = [
  {
    id: 0,
    title: 'Popular',
  },
  {
    id: 1,
    title: 'Price: low to high',
  },
  {
    id: 2,
    title: 'Price: high to low',
  },
  {
    id: 3,
    title: 'Top rated first',
  }];

export { menuUpArray };
