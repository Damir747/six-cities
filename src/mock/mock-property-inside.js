const propertyInside = [
  {
    id: 0,
    title: 'Wi-Fi'
  },
  {
    id: 1,
    title: 'Washing machine'
  },
  {
    id: 2,
    title: 'Towels'
  },
  {
    id: 3,
    title: 'Heating'
  },
  {
    id: 4,
    title: 'Coffee machine'
  },
  {
    id: 5,
    title: 'Baby seat'
  },
  {
    id: 6,
    title: 'Kitchen'
  },
  {
    id: 7,
    title: 'Dishwasher'
  },
  {
    id: 8,
    title: 'Cabel TV'
  },
  {
    id: 9,
    title: 'Fridge'
  }
];

export default propertyInside;
