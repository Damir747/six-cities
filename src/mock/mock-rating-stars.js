const stars = [
  {
    id: 5,
    value: 5,
    title: 'perfect'
  },
  {
    id: 4,
    value: 4,
    title: 'good'
  },
  {
    id: 3,
    value: 3,
    title: 'not bad'
  },
  {
    id: 2,
    value: 2,
    title: 'badly'
  },
  {
    id: 1,
    value: 1,
    title: 'terribly'
  },
];

export default stars;
