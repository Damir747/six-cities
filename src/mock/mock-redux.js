import { redirect } from "../store/middleware/redirect";

const mockRedux = () => {
  const store = {
    getState: jest.fn(() => ({})),
    dispatch: jest.fn(),
  };
  const next = jest.fn();
  const invoke = (action) => redirect(action)(next)(action);
  return { store, next, invoke };
};

export { mockRedux };
