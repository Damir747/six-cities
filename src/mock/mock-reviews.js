const reviews = [{
  id: 0,
  author: 'Max',
  img: 'img/avatar-max.jpg',
  reviewText: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.',
  rating: 80,
  date: 'April 2019'
}];

export default reviews;
