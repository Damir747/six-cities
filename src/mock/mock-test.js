import cities from '../mock/mock-cities';
import propertyInside from '../mock/mock-property-inside';
import { AuthorizationStatus } from '../const';

const initialMockState = {
  HOTEL: {
    hotel: null,
    rooms: [{
      bookmark: "To bookmarks",
      card: "Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.",
      city: { name: "Brussels", location: { latitude: 50.846557, longitude: 4.351697, zoom: 13 } },
      cityName: "Brussels",
      coordinates: { lat: 50.827557, lng: 4.336697, zoom: 16 },
      description: "Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.",
      features: [],
      goods: ["Laptop friendly workspace", "Breakfast"],
      host: { title: "Meet the host", user: {} },
      id: 1,
      images: [],
      img: "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
      level: "",
      priceText: "night",
      priceValue: "939",
      rating: 76,
      title: "Loft Studio in the Central Area",
      type: "house",
    }],
    neighbourhood: [{
      city: {},
      images: [],
      title: '',
      rating: '',
      type: '',
      goods: [],
      host: {},
      description: '',
      id: 2,
      level: '',
      img: '',
      priceValue: '',
      priceText: '',
      bookmark: 'To bookmarks',
      card: '',
    }],
    isHotelListLoading: false,
    isHotelListLoaded: false,
    reviews: [],
    isHotelLoading: false,
    isHotelLoaded: false,
    isCommentLoading: false,
    isCommentLoaded: false,
    isNeighbourhoodLoading: false,
    isNeighbourhoodLoaded: false,
  },
  CITY: {
    activeCity: 'Brussels',
    cities,
    currentCity: 'Brussels',
    isCityListLoaded: false,
    isCityListLoading: false,
  },
  FAVORITES: {
    favorites: [{
      bookmark: "To bookmarks",
      card: "Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.",
      city: { name: "Brussels", location: { latitude: 50.846557, longitude: 4.351697, zoom: 13 } },
      cityName: "Brussels",
      coordinates: { lat: 50.827557, lng: 4.336697, zoom: 16 },
      description: "Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.",
      features: [],
      goods: ["Laptop friendly workspace", "Breakfast"],
      host: { title: "Meet the host", user: {} },
      id: 1,
      images: [],
      img: "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
      level: "",
      priceText: "night",
      priceValue: "939",
      rating: 76,
      title: "Loft Studio in the Central Area",
      type: "house",
    }],
    isFavoriteListLoading: false,
    isFavoriteListLoaded: false,
  },
  COMMENT: {
    comment: '',
  },
  INIT: {
    propertyInside,
  },
  LOGIN: {
    loginName: '',
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  },
  NOTIFICATION: {
    notifications: [],
  },
  SORT: {
    sort: 0,
  }
};

export { initialMockState };
