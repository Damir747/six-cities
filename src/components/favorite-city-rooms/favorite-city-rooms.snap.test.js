import React from 'react';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { initialMockState } from '../../mock/mock-test';
import FavoriteCityRooms from './favorite-city-rooms';

const mockStore = configureStore({});
it(`Should FavoriteCityRooms (snapshot) render correctly`, () => {
  const history = createMemoryHistory();
  const store = mockStore(initialMockState);
  const { container } = render(
    <Provider store={store}>
      <Router history={history}>
        <FavoriteCityRooms favoriteRooms={initialMockState.FAVORITES.favorites} />
      </Router>
    </Provider>
  );

  expect(container).toMatchSnapshot();
});
