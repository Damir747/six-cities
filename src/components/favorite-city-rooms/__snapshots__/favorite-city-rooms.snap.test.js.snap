// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoriteCityRooms (snapshot) render correctly 1`] = `
<div>
  <article
    class="favorites__card place-card"
    data-testid="favorite-city-room-id-1"
  >
    <div
      class="favorites__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
      >
        <img
          alt="Place image"
          class="place-card__image"
          height="110"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg"
          width="150"
        />
      </a>
    </div>
    <div
      class="favorites__card-info place-card__info"
    >
      <div
        class="place-card__price-wrapper"
      >
        <div
          class="place-card__price"
        >
          <b
            class="place-card__price-value"
          >
            €
            939
          </b>
          <span
            class="place-card__price-text"
          >
            / 
            night
          </span>
        </div>
        <button
          class="place-card__bookmark-button button "
          type="button"
        >
          <svg
            class="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlink:href="#icon-bookmark"
            />
          </svg>
          <span
            class="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        class="place-card__rating rating"
      >
        <div
          class="place-card__stars rating__stars"
        >
          <span
            style="width: 80%;"
          />
          <span
            class="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        class="place-card__name"
      >
        <a
          href="/offer/1"
        >
          Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.
        </a>
      </h2>
      <p
        class="place-card__type"
      >
        House
      </p>
    </div>
  </article>
</div>
`;
